### PR TITLE
New Actor for JPET use

### PR DESCRIPTION
**Pull request by BlurredChoise**
Here is working code - have fun.

### CODE REVIEW

**Comment by BlurredChoise** [comment]
This Actor support control of:
- saving particle position along axes X, Y, Z
- saving particle name
- saving data for chosen particles e.g. e+, gamma, etc.
- saving trackID
- saving eventID
- saving parentID
- saving runID
- saving particle kinetic energy at the moment
- saving process name connected with particle at the moment e.g. ionization, compton, etc.
- saving particle's emission point position along axes X,Y,Z
- saving particle primary energy

If you want to use it, create a file **Actor.mac** and include:
`/gate/actor/addActor JPETActor jpet

/gate/actor/jpet/save output/[your file name].root
/gate/actor/jpet/attachTo [Your volume or phantom name]
/gate/actor/jpet/enableXPosition YES
/gate/actor/jpet/enableYPosition YES
/gate/actor/jpet/enableZPosition YES
/gate/actor/jpet/enableParticleName YES
/gate/actor/jpet/enableParticle [particle sign e.g. e+, e-, gamma]
/gate/actor/jpet/enableTrackID YES
/gate/actor/jpet/enableEventID YES
/gate/actor/jpet/enableParentID YES
/gate/actor/jpet/enableRunID YES
/gate/actor/jpet/enableEkin YES
/gate/actor/jpet/enableInteractionTime YES
/gate/actor/jpet/enableProcessName YES
/gate/actor/jpet/enableEmissionPoint YES
/gate/actor/jpet/enablePrimaryEnergy YES`

**ATTENTION!** To off one or more options just set **NO**.

**Everything has been checked, compiled and tested**

File: source/digits_hits/include/GateJPETActor.hh [hunk 1]
```diff
@@ -0,0 +1,197 @@
...
+#include "TString.h"
+
+#include "GateVActor.hh"
+#include "GateJPETActorMessenger.hh"
+
+struct iaea_header_type;
+struct iaea_record_type;
+
+/**
+ * Actor created for JEPT use
```
Comment (wkrzemien) [inline source/digits_hits/include/GateJPETActor.hh]: JEPT -> JPET

File: source/digits_hits/include/GateJPETActor.hh [hunk 2]
```diff
@@ -0,0 +1,197 @@
...
+	// Constructs the sensor
+	virtual void Construct();
+	// Callbacks
+	virtual void UserSteppingAction(const GateVVolume *, const G4Step*);
+	virtual void PreUserTrackingAction(const GateVVolume *, const G4Track*);
+	virtual void BeginOfEventAction(const G4Event * e);
+
+	/// Saves the data collected to the file
+	virtual void SaveData();
+	virtual void ResetData();
```
Comment (wkrzemien) [inline source/digits_hits/include/GateJPETActor.hh:44]: Please, add a comment what ResetData() is for. SaveData() I can guess, but ResetData ?

File: source/digits_hits/include/GateJPETActor.hh [hunk 3]
```diff
@@ -0,0 +1,197 @@
...
+	void SetYPositionEnabled(bool enableYPosition);
+
+	/**
+	 * Record data about z - position
+	 * @param: enableZPosition - set true if u want have data of particle z-position
+	 */
+	void SetZPositionEnabled(bool enableZPosition);
+
+	/**
+	 * Redcord data about particle Name
```
Comment (wkrzemien) [inline source/digits_hits/include/GateJPETActor.hh]: Record

File: source/digits_hits/include/GateJPETActor.hh [hunk 4]
```diff
@@ -0,0 +1,197 @@
...
+	bool EnableInteractionTime;
+	/*Record process name*/
+	bool EnableProcessName;
+	/*Record emission point*/
+	bool EnableEmissionPoint;
+	/*Record primary energy*/
+	bool EnablePrimaryEnergy;
+
+	//Values
+	double positionX;
```
Comment (wkrzemien) [inline source/digits_hits/include/GateJPETActor.hh]: The convention in GATE for members of any class: mKineticEnergy 
or mEnablePrimaryEnergy;

File: source/digits_hits/include/GateJPETActorMessenger.hh [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+#include "GateActorMessenger.hh"
+
+class G4UIcmdWithABool;
+class G4UIcmdWithoutParameter;
+class G4UIcmdWithADoubleAndUnit;
+class G4UIcmdWithAString;
+class GateJPETActor;
+
+/**
+ * Actor messenger created for JEPT use
```
Comment (wkrzemien) [inline source/digits_hits/include/GateJPETActorMessenger.hh]: JPET

File: source/digits_hits/src/GateJPETActor.cc [hunk 6]
```diff
@@ -0,0 +1,257 @@
...
+GateJPETActor::~GateJPETActor()
+{
+	GateDebugMessageInc("Actor", 4, "~GateJPETActor() -- begin" << G4endl);
+	delete pMessenger;
+	GateDebugMessageDec("Actor", 4, "~GateJPETActor() -- end" << G4endl);
+}
+
+void GateJPETActor::Construct()
+{
+	G4cout << "Wywolano JpetActor construct" << G4endl;
```
Comment (wkrzemien) [inline source/digits_hits/src/GateJPETActor.cc]: in English pls :)

File: source/digits_hits/src/GateJPETActor.cc [hunk 7]
```diff
@@ -0,0 +1,257 @@
...
+#include "GateSourceMgr.hh"
+#include "G4ParticleTable.hh"
+
+GateJPETActor::GateJPETActor(G4String name, G4int depth):GateVActor(name, depth)
+{
+	GateDebugMessageInc("Actor", 4, "GateJPETActor() -- begin" << G4endl);
+	pMessenger = new GateJPETActorMessenger(this);
+
+	//For default set everything for FALSE
+	 EnableXPosition = false;
```
Comment (wkrzemien) [inline source/digits_hits/src/GateJPETActor.cc]: with m

**Pull request merged**